lib/language_server/protocol/interface/base.rb

```diff
@@ -0,0 +1,68 @@
+module LanguageServer
+  module Protocol
+    module Interface
+      class Base
+        def self.define_attribute_method(key)
+          lower_camel_case = Protocol::Utils.to_lower_camerize(key.to_s)
+
+          class_eval(<<-METHOD, __FILE__, __LINE__ + 1)
+            def #{key}; @attributes[:#{lower_camel_case}] end
```

Comment:
Use `#[]` instead of `#fetch` because instance's attributes are already validated.